scopa.py

```diff
@@ -0,0 +1,217 @@
+    def __init__(self):
+        self.generate_deck()
+        # self.deck = ['03 ', '06 ', '05 ', '12 ', '01 ', '12*', '05*', '03 ', '02 ', '03 ', '11*', '05 ', '04 ', '03*', '01 ', '04*', '06*', '04 ', '02 ', '07 ', '13 ', '13*', '01*', '01 ', '07 ', '07*', '11 ', '07 ', '02 ', '12 ', '13 ', '06 ', '12 ', '11 ', '06 ', '04 ', '13 ', '02*', '05 ', '11 ']
+        self.initiate_play()
+
+    # this method generates random deck
+    def generate_deck(self):
+        all_cards = []
+        for color in colors:
+            for i in range(1, 8):
+                card = "0" + str(i) + color
+                all_cards.append(card)
+            for j in range(11, 14):
+                card = str(j) + color
+                all_cards.append(card)
+
+        self.deck = []
+        while len(all_cards) > 0:
+            pos = random.randint(0, len(all_cards) - 1)
+            random_card = all_cards[pos]
+            all_cards.remove(random_card)
+            self.deck.append(random_card)
+
+        print (self.deck)
+
+    # create table, hands, piles and scopa count
+    def initiate_play(self):
+        for i in range(0, initial_table_size):
+            self.table.append(self.deck.pop())
+
+        for p in range(0, no_of_players):
+            hand = []
+            self.draw_hand(hand)
+            self.hands.append(hand)
+            self.scopa_count.append(0)
+            self.piles.append([])
+
+    # draw cards from deck to hand, return False if hand is empty
+    def draw_hand(self, hand):
+        if len(self.deck) == 0:
+            return False
+        for c in range(0, hand_size):
+            hand.append(self.deck.pop())
+        return True
+
+    def print_game_results(self):
+        print("Scopas: " + str(self.scopa_count))
+        # calculate results
+        for s in range(0, no_of_players):
+            pile_score = self.calculate_score(s)
+
+            print("Score for hand " + str(s) + ", cards " + str(len(self.piles[s])) + " ("
+                  + str(self.piles[s]) + ") is " + str(pile_score))
+
+    def print_status(self):
+        print("===========")
+        print("Table:")
+        print(self.table)
+        print("Hands")
+        print(self.hands)
+        print("Deck")
+        print(self.deck)
+        print("Scopas")
+        print(self.scopa_count)
+        for pile in self.piles:
+            print("Pile size: "+str(len(pile))+", content: "+str(pile))
+        print("===========\n")
+
+    def check_sum_of_cards(self):
+        csum = len(self.table)
+        csum += len(self.deck)
+        for hand in self.hands:
+            csum += len(hand)
+        for pile in self.piles:
+            csum += len(pile)
+
+        if csum != 40:
+            print("Sum is "+str(sum)+" instead of 40")
+            exit(1)
+
+    # returns possible takes - each item in the list is a couple: card from hand, cards to be taken
+    def possible_takes(self, hand):
+        takes_with_sum = tactics.all_takes_with_sum(self.table)
```

Comment:
Remove this variable as it's used only once and doesn't provide any new information.